### ISSUE TITLE
pool=true turns string into missing value

### ISSUE THREAD

**Issue by nalimilan** [JuliaData/CSV.jl#402]
`pool=true` can make a cell contain a missing value instead of a (long and weird) string:
```
julia> typeof.(eachcol(CSV.read("test/testfiles/test_file_issue_154.csv", normalizenames=true)))
4-element Array{DataType,1}:
 Array{Int64,1}
 Array{Int64,1}
 WeakRefStrings.StringArray{String,1}
 WeakRefStrings.StringArray{String,1}

julia> ismissing.(CSV.read("test/testfiles/test_file_issue_154.csv", normalizenames=true)[4])
2-element BitArray{1}:
 false
 false

julia> typeof.(eachcol(CSV.read("test/testfiles/test_file_issue_154.csv", normalizenames=true, pool=true)))
4-element Array{DataType,1}:
 Array{Int64,1}
 Array{Int64,1}
 PooledArrays.PooledArray{String,UInt32,1,Array{UInt32,1}}
 PooledArrays.PooledArray{Union{Missing, String},UInt32,1,Array{UInt32,1}}

julia> ismissing.(CSV.read("test/testfiles/test_file_issue_154.csv", normalizenames=true, pool=true)[4])
2-element BitArray{1}:
 false
  true
```

**Comment by quinnj** [comment]
I don't see the same behavior on master:
```julia
julia> df2.Column4
2-element PooledArrays.PooledArray{Union{Missing, String},UInt32,1,Array{UInt32,1}}:
 " comment "
 missing    

julia> typeof.(eachcol(df))
4-element Array{DataType,1}:
 Array{Int64,1}                                      
 Array{Int64,1}                                      
 WeakRefStrings.StringArray{String,1}                
 WeakRefStrings.StringArray{Union{Missing, String},1}

julia> typeof.(eachcol(df2))
4-element Array{DataType,1}:
 Array{Int64,1}                                                           
 Array{Int64,1}                                                           
 PooledArrays.PooledArray{String,UInt32,1,Array{UInt32,1}}                
 PooledArrays.PooledArray{Union{Missing, String},UInt32,1,Array{UInt32,1}}
```

**Comment by quinnj** [comment]
Ok, I can reproduce when `pool=0.1`, but not when `pool=true`. I pushed a fix to your #403 branch. Definitely a tricky case; it only occurs when we simultaneously discover that a pooled column should no longer be treated as pooled (# of unique values is over the pool threshold) and also detect the first `missing` value in that column. Luckily, it's something that would probably occur in testing (as we encountered) rather than real world.
